sql/stats: TestDeleteFailedJob failed

sql/stats.TestDeleteFailedJob [failed](https://tanzanite.cluster.engflow.com/invocations/default/389e6840-3599-499a-b0e2-9cd00230a383?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [e2e88aa288c81282b20ca0640da983f97b234324](https://github.com/cockroachdb/cockroach/commits/e2e88aa288c81282b20ca0640da983f97b234324):


```
=== RUN   TestDeleteFailedJob
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDeleteFailedJob3850959274
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    create_stats_job_test.go:332: one job should have succeeded but both failed. err1:pq: another CREATE STATISTICS job is already running, err2:pq: another CREATE STATISTICS job is already running
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestDeleteFailedJob3850959274
--- FAIL: TestDeleteFailedJob (175.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDeleteFailedJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34818

--- a/pkg/server/application_api/schema_inspection_test.go
+++ b/pkg/server/application_api/schema_inspection_test.go
@@ -482,8 +482,7 @@ func TestAdminAPITableStats(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderDeadlockWithIssue(t, 115914, "likely to time out")
-
+	skip.UnderDeadlock(t, "low ScanMaxIdleTime and deadlock overloads the EngFlow executor")
 	skip.UnderStress(t, "flaky under stress #107156")
 	skip.UnderRace(t, "flaky under race #107156")
 

--- a/pkg/sql/stats/delete_stats_test.go
+++ b/pkg/sql/stats/delete_stats_test.go
@@ -673,6 +673,7 @@ func TestStatsAreDeletedForDroppedTables(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t) // slow test
+	skip.UnderDeadlock(t, "low ScanMaxIdleTime and deadlock overloads the EngFlow executor")
 
 	var params base.TestServerArgs
 	params.ScanMaxIdleTime = time.Millisecond // speed up MVCC GC queue scans

